### PR TITLE
jdk11 Post-sprint activities

### PR DESCRIPTION
**Pull request by tbarsballe**
jdk11 Post-sprint activities, per the [spreadsheet](https://docs.google.com/spreadsheets/d/1oE6mU4jp-ZL5PebgXf-fuhtf7MY5dzSwPqpMtrzdZ94/edit#gid=2055024842)

* [x] org.geotools.data.complex.ComplexFeatureConstraints
* [x] org.geotools.data.complex.filter.XPathUtil moved to app-schema
* [x] org.geotools.data.complex.config
* [ ] org.sqlite
* [x] java 11 build instructions and warning

Currently WIP, waiting to merge until everything is included

### CODE REVIEW

File: docs/user/build/install/jdk.rst [hunk 1]
```diff
@@ -48,23 +41,37 @@ Java 5    GeoTools 2.5.x   GeoTools 8.x     compiler=1.5
...
+If your project depends on GeoTools 21.x, and you want to use it with Java 11:
 
-Java 9 introduces a number of changes to the JVM, most notably the module system (Project Jigsaw). Refer to `The State of the Module System <http://openjdk.java.net/projects/jigsaw/spec/sotms/>`_ for more details.
-While GeoTools has not yet been updated to use the module system, GeoTools 19.x can run on Java 9.
+* Use the GeoTools jars on the module path, they are configured with automatic module names
 
-If your project depends on GeoTools 19.x, and you want to use it with Java 9, you will need to add the following flags to your JVM runtime arguments::
+* Use the GeoTools jars on the classpath, adding the following flags to your JVM runtime arguments::
 
     --add-modules=java.xml.bind --add-modules=java.activation
```
Comment (tbarsballe) [inline docs/user/build/install/jdk.rst]: This shouldn't be necessary anymore; we have these modules as proper dependencies now

File: docs/user/build/install/jdk.rst [hunk 2]
```diff
@@ -48,23 +41,37 @@ Java 5    GeoTools 2.5.x   GeoTools 8.x     compiler=1.5
...
 
-This adds the JAXB and Activation modules to the Java Runtime (They are not included by default in Java 9).
+  These arguments are not supported under Java 8, to retain compatibility with Java 8 include::
 
-Note that these arguments are not supported under Java 8. If you wish to retain compatibility with Java 8, also include::
+    --add-modules=java.xml.bind --add-modules=java.activation -XX:+IgnoreUnrecognizedVMOptions
 
-    -XX:+IgnoreUnrecognizedVMOptions
+  This tells Java 8 to ignore options it doesn't recognize and not throw an error.
```
Comment (tbarsballe) [inline docs/user/build/install/jdk.rst]: With the changed lines above, this is now innacurate

File: docs/user/build/install/jdk.rst [hunk 3]
```diff
@@ -48,23 +41,37 @@ Java 5    GeoTools 2.5.x   GeoTools 8.x     compiler=1.5
...
 
-This tells the JVM to ignore options it doesn't recognize, so Java 8 won't throw an error.
+.. note::
+   
+   Compatibility note due to breaking API changes in Java 11:
+   
+   * Building with Java 8, compatible with Java 8 and Java 11.
+   * Building with Java 11, compatible with Java 11.
```
Comment (tbarsballe) [inline docs/user/build/install/jdk.rst]: This could be reworded to be a bit clearer

File: docs/user/build/install/jdk.rst [hunk 4]
```diff
@@ -105,20 +112,17 @@ One technique is to set up a batch file similar to the following:
    
    Placing JAVA_HOME on the path before System32 shortcuts this annoying "feature".
 
-Java Extensions (Optional)
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+Java Extensions (Pre Java 9)
```
Comment (tbarsballe) [inline docs/user/build/install/jdk.rst]: Everywhere else references to Java 9 have been removed and replaced with Java 11.
Perhaps use "Java 8 and before" instead, for consistency?

File: docs/user/build/install/jdk.rst [hunk 5]
```diff
@@ -105,20 +112,17 @@ One technique is to set up a batch file similar to the following:
...
-Java Developers Kit:
-* Can make use of JAI and ImageIO if they have been installed into your JDK
-Java Runtime Environment:
-* Can make use of JAI and ImageIO if you have installed them into your JRE
+========================== ========== ==========
+Installation               JDK        JRE
+========================== ========== ==========
+JDK Installation           Available  Available
+JRE Installation                      Available
+========================== ========== ==========
```
Comment (tbarsballe) [inline docs/user/build/install/jdk.rst]: This is really unclear and seems to have the same values on both the X and Y axes. Did you mean to include JAI and ImageIO somewhere in this table?

File: docs/user/build/install/jdk.rst [hunk 6]
```diff
@@ -234,8 +238,6 @@ The goal is to place the required jars into your lib/ext directory of both your
 Optional: Mac ImageIO
 '''''''''''''''''''''
 
-Java Advanced Imaging is included with recent releases of Mac OS (but that may be changing in the future).
-
 The JAI ImageIO extension is not available as a download for the mac. However, you can use the jar from the Linux/windows download to get “pure java” functionality without hardware acceleration:
```
Comment (tbarsballe) [inline docs/user/build/install/jdk.rst]: Capital M for mac

File: docs/user/build/install/jdk.rst (hunk 5 above)
Comment (jodygarnett) [reply docs/user/build/install/jdk.rst]: added it as a block to each section

File: docs/user/build/install/jdk.rst (hunk 6 above)
Comment (jodygarnett) [reply docs/user/build/install/jdk.rst]: changed to macOS

File: docs/user/build/install/jdk.rst (hunk 1 above)
Comment (tbarsballe) [reply docs/user/build/install/jdk.rst]: Yeah, but they aren't part of the JVM anymore, there standalone dependencies, so either way we shouldn't need this runtime argument. GeoTools/GeoServer works fine without it (and it runs on the classpath)

File: modules/extension/complex/src/main/java/org/geotools/data/complex/feature/type/ComplexTypeProxy.java [hunk 7]
```diff
@@ -27,7 +27,7 @@
  * @author Gabriel Roldan (Axios Engineering)
  * @since 2.4
  */
```
Comment (smithkm) [inline modules/extension/complex/src/main/java/org/geotools/data/complex/feature/type/ComplexTypeProxy.java]: Would be nice to add meaningful docs to classes if they are becoming public

Comment (tbarsballe) [reply modules/extension/complex/src/main/java/org/geotools/data/complex/feature/type/ComplexTypeProxy.java]: They were previously relying on package visibility, and can't anymore.

**Pull request merged**